### PR TITLE
Brightroll adapter - adding config support

### PR DESCRIPTION
**Pull request by smithaammassamveettil**
Description:
Adding configuration support through extraAdapterInfo for Brightroll adapter
Moving existing publisher adapter configuration in the code to configuration

@SyntaxNode

### CODE REVIEW

File: config/config.go
Comment (SyntaxNode) [inline config/config.go:862]: I'd just like to confirm that this is the default config you'd like to set for all hosts, and is not specific to AppNexus/Xandr.

File: adapters/brightroll/brightroll.go
Comment (smithaammassamveettil) [reply adapters/brightroll/brightroll.go]: Sure. Updated

File: adapters/brightroll/brightroll.go
Comment (smithaammassamveettil) [reply adapters/brightroll/brightroll.go]: Done

File: adapters/brightroll/brightroll.go
Comment (smithaammassamveettil) [reply adapters/brightroll/brightroll.go]: Updated. Thanks!

File: config/config.go
Comment (smithaammassamveettil) [reply config/config.go]: This is the default config to set across all hosts under Appnexus Go Prebid Server, and NOT for Rubicon Java Prebid Server.

**Comment by smithaammassamveettil** [comment]
@SyntaxNode Noticed that the build here is failing on go version 1.13. Build is passing on 1.14. Why building on two versions? Not sure why it is failing on 1.13.

**Comment by SyntaxNode** [comment]
I'm not sure either. I kicked it off again to see if this is a repeating issue.

File: config/config.go
Comment (SyntaxNode) [reply config/config.go]: There are other hosts running Prebid Server other than AppNexus/Xandr. If this config is specifically just for them, I recommend default the extra info to empty in this PR and I'll move the configuration internal to just AppNexus/Xandr.

Comment (smithaammassamveettil) [reply config/config.go]: This configuration is only for publishers we integrated through Appnexus Go Prebid Server. We have 2 publishers currently, when those publisher's bid request comes, these configuration is needed. Those publishers bid requests will come through only Appnexus Go Prebid Server. Please advise how to proceed in that case. Add the configuration in this PR or send to you separately? If it is the second option, what is the procedure to do in future,  if we need to change any existing configuration or add a new configuration?

Comment (SyntaxNode) [reply config/config.go]: > Please advise how to proceed in that case.

Please remove the SetDefault call for the extra info. I'll add that internally to the AppNexus host config to coincide with the release of this PR. In the future, email us at prebid-server@xandr.com with updates. That email alias goes directly to my team.

Comment (smithaammassamveettil) [reply config/config.go]: Ok. removed the SetDefault from code. Please find attached configuration to add internally
`{\"accounts\": [{\"id\": \"adthrive\",\"bcat\": [\"IAB8-5\", \"IAB8-18\", \"IAB15-1\", \"IAB7-30\", \"IAB14-1\", \"IAB22-1\", \"IAB3-7\", \"IAB7-3\", \"IAB14-3\", \"IAB11\", \"IAB11-1\", \"IAB11-2\", \"IAB11-3\", \"IAB11-4\", \"IAB11-5\", \"IAB23\", \"IAB23-1\", \"IAB23-2\", \"IAB23-3\", \"IAB23-4\", \"IAB23-5\", \"IAB23-6\", \"IAB23-7\", \"IAB23-8\", \"IAB23-9\", \"IAB23-10\", \"IAB7-39\", \"IAB9-30\", \"IAB7-44\", \"IAB25\", \"IAB25-1\", \"IAB25-2\", \"IAB25-3\", \"IAB25-4\", \"IAB25-5\", \"IAB25-6\", \"IAB25-7\", \"IAB26\", \"IAB26-1\", \"IAB26-2\", \"IAB26-3\", \"IAB26-4\"],\"battr\": [1,2,3,6,9,10]},
{\"id\": \"dailymail\",\"badv\": [\"wsj.com\",\"thesun.co.uk\",\"marksandspencer.com\",\"nytimes.com\",\"thetimes.co.uk\",\"theguardian.com\",\"huffingtonpost.com\",\"ft.com\",\"bbc.com\",\"telegraph.co.uk\",\"economist.com\",\"s1jobs.com\",\"fisherinvestments.com\",\"independent.co.uk\",\"s1homes.com\",\"newsquestne.co.uk\",\"newsquest.co.uk\",\"chorleycitizen.co.uk\",\"theboltonnews.co.uk\",\"prestwichandwhitefieldguide.co.uk\",\"burytimes.co.uk\",\"theoldhamtimes.co.uk\",\"wirralglobe.co.uk\",\"thenorthernecho.co.uk\",\"darlingtonandstocktontimes.co.uk\",\"asianimage.co.uk\",\"lancashiretelegraph.co.uk\",\"sthelensstar.co.uk\",\"runcornandwidnesworld.co.uk\",\"leighjournal.co.uk\",\"warringtonguardian.co.uk\",\"northwichguardian.co.uk\",\"winsfordguardian.co.uk\",\"knutsfordguardian.co.uk\",\"creweguardian.co.uk\",\"newsandstar.co.uk\",\"hexham-courant.co.uk\",\"nwemail.co.uk\",\"timesandstar.co.uk\",\"cumbrialife.co.uk\",\"thewestmorlandgazette.co.uk\",\"messengernewspapers.co.uk\",\"gazetteherald.co.uk\",\"yorkpress.co.uk\",\"cravenherald.co.uk\",\"keighleynews.co.uk\",\"wharfedaleobserver.co.uk\",\"ilkleygazette.co.uk\",\"thetelegraphandargus.co.uk\",\"herefordtimes.com\",\"ludlowadvertiser.co.uk\",\"redditchadvertiser.co.uk\",\"droitwichadvertiser.co.uk\",\"bromsgroveadvertiser.co.uk\",\"cotswoldjournal.co.uk\",\"tewkesburyadmag.co.uk\",\"eveshamjournal.co.uk\",\"dudleynews.co.uk\",\"stourbridgenews.co.uk\",\"halesowennews.co.uk\",\"kidderminstershuttle.co.uk\",\"malverngazette.co.uk\",\"ledburyreporter.co.uk\",\"worcesternews.co.uk\",\"southendstandard.co.uk\",\"essexcountystandard.co.uk\",\"gazette-news.co.uk\",\"clactonandfrintongazette.co.uk\",\"harwichandmanningtreestandard.co.uk\",\"braintreeandwithamtimes.co.uk\",\"halsteadgazette.co.uk\",\"guardian-series.co.uk\",\"chelmsfordweeklynews.co.uk\",\"thurrockgazette.co.uk\",\"basildonstandard.co.uk\",\"echo-news.co.uk\",\"bucksfreepress.co.uk\",\"theargus.co.uk\",\"romseyadvertiser.co.uk\",\"dailyecho.co.uk\",\"hampshirechronicle.co.uk\",\"basingstokegazette.co.uk\",\"andoveradvertiser.co.uk\",\"stalbansreview.co.uk\",\"watfordobserver.co.uk\",\"heraldseries.co.uk\",\"bicesteradvertiser.net\",\"witneygazette.co.uk\",\"banburycake.co.uk\",\"oxfordmail.co.uk\",\"oxfordtimes.co.uk\",\"brentwoodlive.co.uk\",\"maldonandburnhamstandard.co.uk\",\"falmouthpacket.co.uk\",\"smallholder.co.uk\",\"southwestfarmer.co.uk\",\"dorsetecho.co.uk\",\"bridportnews.co.uk\",\"wiltsglosstandard.co.uk\",\"gazetteseries.co.uk\",\"bridgwatermercury.co.uk\",\"burnhamandhighbridgeweeklynews.co.uk\",\"chardandilminsternews.co.uk\",\"somersetcountygazette.co.uk\",\"thisisthewestcountry.co.uk\",\"yeovilexpress.co.uk\",\"wiltshiretimes.co.uk\",\"swindonadvertiser.co.uk\",\"salisburyjournal.co.uk\",\"campaignseries.co.uk\",\"southwalesguardian.co.uk\",\"milfordmercury.co.uk\",\"pembrokeshirecountyliving.co.uk\",\"tivysideadvertiser.co.uk\",\"westerntelegraph.co.uk\",\"southwalesargus.co.uk\",\"freepressseries.co.uk\",\"barryanddistrictnews.co.uk\",\"penarthtimes.co.uk\",\"eveningtimes.co.uk\",\"scottishfarmer.co.uk\",\"s1cars.com\",\"heraldscotland.com\",\"thenational.scot\",\"boxingnewsonline.net\",\"pensions-insight.co.uk\",\"insurancetimes.co.uk\",\"globalreinsurance.com\",\"strategicrisk-asiapacific.com\",\"strategic-risk-europe.com\",\"thestrad.com\",\"times-series.co.uk\",\"ealingtimes.co.uk\",\"enfieldindependent.co.uk\",\"harrowtimes.co.uk\",\"hillingdontimes.co.uk\",\"borehamwoodtimes.co.uk\",\"thetottenhamindependent.co.uk\",\"newsshopper.co.uk\"],\"bcat\": [\"IAB11\",\"IAB12\",\"IAB13\",\"IAB14\",\"IAB25\",\"IAB26\",\"IAB23\"],\"battr\": [1,2,3,6,7,8,9,10,13,14],\"bidfloor\": 0.05}]}`

Comment (smithaammassamveettil) [reply config/config.go]: Ok. removed the SetDefault from code. Shared below document having the configuration to add interally as you mentioned above to  prebid-server@xandr.com
https://docs.google.com/document/d/14-jMJjBvWaftCNkeN8Dx60eJrwgpaw-CAr5M7mfnESM/edit?usp=sharing

Need to add this configuration first or simultaneously with this code release for existing publishers to work with this code changes.

**Review by SyntaxNode — DISMISSED** [review]

File: config/config.go
Comment (SyntaxNode) [reply config/config.go]: I requested access to the doc. We need to use the string with the escape characters. The Xandr config changes will be released simultaneously with this code.

Comment (smithaammassamveettil) [reply config/config.go]: Access provided. Please check. Thanks.

File: adapters/brightroll/brightroll.go
Comment (mansinahar) [inline adapters/brightroll/brightroll.go:73]: Please consider extracting the account/publisher validation piece of code in its own separate method

File: adapters/brightroll/brightroll.go
Comment (mansinahar) [inline adapters/brightroll/brightroll.go:233]: Please consider adding unit tests for this function

File: adapters/brightroll/brightroll.go
Comment (smithaammassamveettil) [reply adapters/brightroll/brightroll.go:73]: Will do the refactoring code as a separate PR. thanks.

File: adapters/brightroll/brightroll.go
Comment (smithaammassamveettil) [reply adapters/brightroll/brightroll.go:233]: Yes, there are testcases (existing testcases) updated to cover this function

**Comment by smithaammassamveettil** [comment]
@SyntaxNode  When will config changes and this code changes available in production?

**Comment by SyntaxNode** [comment]
> @SyntaxNode When will config changes and this code changes available in production?

This PR needs a second approval before it can merge in. Once that's done, the next release is will likely either 9/9 or 9/10.

File: adapters/brightroll/brightroll.go
Comment (SyntaxNode) [reply adapters/brightroll/brightroll.go:233]: This is what the test coverage for `NewBrightrollBidder` looks like. @mansinahar, what were you looking for coverage on?

Screen Shot 2020-09-03 at 1 49 52 PM

Comment (mansinahar) [reply adapters/brightroll/brightroll.go:233]: I believe the existing tests i.e. the `RunJSONBidderTest` are more integration tests for the entire adapter code. I was more suggesting to add unit tests for this particular `NewBrightrollBidder` method as this is no longer a simple constructor that takes parameters and assigns them to the object. I'd expect this to be a simple test consisting of two test cases, one that calls the `NewBrightrollBidder` function with an empty string for `extraAdapterInfo` and another that calls it with a non-empty, valid `extraAdapterInfo` string.

Comment (smithaammassamveettil) [reply adapters/brightroll/brightroll.go:233]: @mansinahar Can you please provide some guidance on where/how to add such testcases in the existing adapter framework or please point me to any such existing testcases to refer.

**Comment by smithaammassamveettil** [comment]
@mansinahar  Can you please help to review the changes. Added testcases as you suggested.

@SyntaxNode  Looks like your review got dismissed when I added testcases. Can you please help re-review/approve again?

We would need this change in the next release asap.
Thanks
Smitha

**Review by mansinahar — APPROVED** [review]

**Review by SyntaxNode — APPROVED** [review]

**Comment by smithaammassamveettil** [comment]
Thanks for approving and merging.
@SyntaxNode Can you please let me know when are these changes (code & config setup) going to release?

**Comment by smithaammassamveettil** [comment]
@SyntaxNode or @mansinahar  Is this PR and related config setup released to production? Please let me know. thanks.

**Pull request merged**
